client/entrypoints/express-checkout/index.js

```diff
@@ -621,7 +621,10 @@ jQuery( function ( $ ) {
 	};
 
 	// We don't need to initialize ECE on the checkout page now because it will be initialized by updated_checkout event.
-	if ( ! getExpressCheckoutData( 'is_checkout_page' ) ) {
+	if (
+		! getExpressCheckoutData( 'is_checkout_page' ) ||
+		getExpressCheckoutData( 'is_pay_for_order' )
```

Comment:
🔖 Noting that I had to add this logic otherwise the ECE buttons weren't initialised on the pay for order page. 

Pay for Order pages are considered checkout pages and so the original `! getExpressCheckoutData( 'is_checkout_page' )` logic prevented the buttons being initialized.

Reply:
@james-allan please add this note to the comment above. 
I removed this check last time as I was unaware of this reason. These two conditions looked redundant together when I did not know the reason.

Reply:
TBH, adding this check boggled my mind too because it seems so odd. Call `init` if not checkout OR pay for order is confusing. I wonder if it would be worth updating this logic to be more explicit rather than exclusionary. 

ie list the pages we do want to init on, rather than saying all pages except the checkout.